ophyd/areadetector/plugins.py

```diff
@@ -59,14 +59,30 @@ def register_plugin(cls):
 
 class PluginBase(ADBase):
     '''AreaDetector plugin base class'''
-    def __init__(self, *args, **kwargs):
+    def __init__(self, *args, configuration_attrs=None, **kwargs):
+        if configuration_attrs is None:
+            configuration_attrs = self._default_configuration_attrs
         # Turn array callbacks on during staging.
         # Without this, no array data is sent to the plugins.
-        super().__init__(*args, **kwargs)
-
+        super().__init__(*args, configuration_attrs=configuration_attrs,
+                         **kwargs)
+        # make sure it is the right type of plugin
+        if not self.plugin_type.get().startswith(self._plugin_type):
```

Comment:
It may be useful to remove this check in some cases. I should be able to use `PluginBase` with any plugin and not always the exact plugin class.

To that end, can you add a `if self._plugin_type is not None and ...` before this?